arch/riscv/core/thread.c

```diff
@@ -212,6 +212,8 @@ FUNC_NORETURN void z_riscv_switch_to_main_no_multithreading(k_thread_entry_t mai
 	main_stack = (K_THREAD_STACK_BUFFER(z_main_stack) +
 		      K_THREAD_STACK_SIZEOF(z_main_stack));
 
+	irq_unlock(MSTATUS_IEN);
```

Comment:
Can't wrap my head around the reason for enabling the interrupt and jump to the user code. Please, prove me wrong that this isn't a bad practice!

Reply:
I understand that after the explanations from #74177, everything is clear now?

Reply:
hi @jaz1-nordic yep, this exposed an issue with re-enabling the clock in the application code. All good now!